packages/services/f-analytics/src/store/analytics.module.js

```diff
@@ -25,24 +32,58 @@ export default {
             globalUserId: undefined,
             signinType: undefined,
             signupDate: undefined
-        }
+        },
+        events: []
     }),
 
     actions: {
-        updatePlatformData: ({ commit }, platformData) => {
-            commit('pushPlatformData', platformData);
+        pushPlatformData: ({ commit, state }, platformData) => {
+            if (platformData) {
+                commit(PUSH_PLATFORM_DATA, platformData);
+            }
+
+            if (isDataLayerPresent()) {
+                window.dataLayer.push({ platformData: { ...state.platformData } });
+            }
+        },
+
+        pushUserData: ({ commit, state }, userData) => {
+            if (userData) {
+                commit(PUSH_USER_DATA, userData);
+            }
+
+            if (isDataLayerPresent()) {
+                window.dataLayer.push({ userData: { ...state.userData } });
+            }
         },
-        updateUserData: ({ commit }, userData) => {
-            commit('pushUserData', userData);
+
+        pushEvent: ({ commit, state }, event) => {
+            if (event) {
+                commit(PUSH_EVENT, event);
+            }
+
+            if (isDataLayerPresent() && state.events) {
+                state.events.forEach(e => window.dataLayer.push({ ...e }));
+                commit(CLEAR_EVENTS);
+            }
         }
     },
 
     mutations: {
-        pushPlatformData: (state, platformData) => {
-            state.platformData = platformData;
+        [PUSH_PLATFORM_DATA]: (state, platformData) => {
+            state.platformData = { ...state.platformData, ...platformData };
         },
-        pushUserData: (state, userData) => {
+
+        [PUSH_USER_DATA]: (state, userData) => {
             state.userData = userData;
```

Comment:
This should be also changed:
`state.userData = { ...state.userData, ...userData };`